subsys/bluetooth/enocean.c

```diff
@@ -378,13 +401,15 @@ static void adv_recv(const struct bt_le_scan_recv_info *info,
 	uint8_t len = net_buf_simple_pull_u8(buf);
 	uint8_t type = net_buf_simple_pull_u8(buf);
 	bool has_shortened_name = (type == BT_DATA_NAME_SHORTENED);
+	/* Upper two bytes of address indicate device type */
+	uint16_t device_type = sys_get_le16(&info->addr->a.val[4]);
```

Comment:
This is not a full proof method. This will only work for commissioning telegrams. But for data telegrams, if the devices are configured to transmit using RPA, module won't be able to infer device type.

We may have to refactor this module later to properly support RPA based data telegrams.

Reply:
I thought the controller resolved the RPA and passed it pre-resolved to this? If that's not the case, we need bigger changes, since the address is used to lookup the key when switch data is received. If you'd like to check this, go ahead, and create a separate issue for supporting RPA switches if it doesn't already work as-is.

Reply:
(I wasn't able to check this since the PTM-215B I have available apparently can't be reconfigured using the Android App, but needs a PC-based NFC reader/writer)

Reply:
That is not how RPA works. After controller resolves RPA, it will inform the "identity address" corresponding to this RPA. This "identity" can be anything user defined. My impression is that, this requires some amount of refactoring to support RPA based data telegrams (commissioning telegrams are never sent using RPA). Needs a seperate task.